backend/src/uruguay_news/services.py

```
@@ -0,0 +1,162 @@
+    """
+    Extracts and preprocesses content from news articles.
+
+    Handles URL-based extraction, text cleaning, and content validation
+    for downstream AI analysis.
+    """
+
+    def __init__(self) -> None:
+        """Initialize the content extractor."""
+        self.logger = logging.getLogger(f"{__name__}.{self.__class__.__name__}")
+
+    async def extract_from_url(self, url: str) -> Optional[Dict[str, Any]]:
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)